src/xtopcom/xdata/src/xdata_structures.cc

```diff
@@ -398,6 +398,10 @@ bool xreg_node_info::can_be_exchange() const noexcept {
 }
 
 bool xreg_node_info::can_be_fullnode() const noexcept {
+    return (could_be_auditor() && genesis());
```

Comment:
`return could_be_fullnode() && genesis();`

check to see if `could_be_fullnode()` works for us or not.